### PR TITLE
Enhanced client.py options

### PR DESCRIPTION
**Pull request by Polochon-street**

### CODE REVIEW

File: mpd/client.py [hunk 1]
```diff
@@ -217,168 +217,93 @@ def _init():
 
     return client, conn, cur, current_song_coords
 
-
```
Comment (Phyks) [inline mpd/client.py]: There should be two lines between functions (PEP8).

File: mpd/client.py [hunk 2]
```diff
@@ -217,168 +217,93 @@ def _init():
...
-                    closest_album = current_album_set
-
-        logging.info("Closest album found is \"%s\". Distance is %f." % (closest_album[0]["album"], shortest_distance))
-        for song in closest_album:
+        # Get all albums
+        cur.execute("SELECT DISTINCT album FROM songs")
+        albums = cur.fetchall();
+
+        # Compute the distance between current album and all other albums
+        for tmp_album in albums:
```
Comment (Phyks) [inline mpd/client.py:240]: Better to use `for tmp_album in cur.execute()` which uses an iterator. See 5th example on https://docs.python.org/2/library/sqlite3.html.

File: mpd/client.py [hunk 3]
```diff
@@ -217,168 +217,93 @@ def _init():
 
     return client, conn, cur, current_song_coords
 
-
-def main_album(queue_length):
+def main_album(queue_length, option_best):
```
Comment (Phyks) [inline mpd/client.py]: Maybe we can use `option_best=True` to keep the previous behavior as default for this function?

File: mpd/client.py [hunk 4]
```diff
@@ -217,168 +217,93 @@ def _init():
...
             client.add(song["filename"])
-        current_song_coords = closest_album[-1]
 
     conn.close()
     client.close()
     client.disconnect()
 
 
-def main_single(queue_length):
+def main_single(queue_length, option_best):
```
Comment (Phyks) [inline mpd/client.py]: Same for `option_best`.

File: mpd/server.py [hunk 5]
```diff
@@ -85,25 +85,30 @@ def full_rescan(mpd_root):
...
+        subprocess.check_call(["blissify", mpd_root] + all_songs[i*100:(i+1)*100])
+    subprocess.check_call(["blissify", mpd_root] + all_songs[len(all_songs) - (len(all_songs) % 100):len(all_songs)])
+
     # Update the latest mtime stored
     latest_mtime = 0
     try:
         with open(os.path.join(_BLISSIFY_DATA_HOME, "latest_mtime.txt"),
                   "r") as fh:
-            latest_mtime = int(fh.read())
+            latest_mtime = int(str(fh.read()))
```
Comment (Phyks) [inline mpd/server.py]: What is the reason for this change?

File: mpd/server.py [hunk 6]
```diff
@@ -85,25 +85,30 @@ def full_rescan(mpd_root):
...
+
     for song in all_songs:
-        last_modified = client.find("file", song)["last_modified"]
+        last_modified = client.find("file", song)[0]["last-modified"]
         last_modified = int(dateutil.parser.parse(last_modified).timestamp())
         if last_modified > latest_mtime:
             latest_mtime = last_modified
     with open(os.path.join(_BLISSIFY_DATA_HOME, "latest_mtime.txt"), "w") as fh:
-        fh.write(latest_mtime)
+        print(latest_mtime)
```
Comment (Phyks) [inline mpd/server.py]: Did you forget to delete this print statement?

File: mpd/server.py [hunk 7]
```diff
@@ -85,25 +85,30 @@ def full_rescan(mpd_root):
...
+        fh.write(str(latest_mtime))
```
Comment (Phyks) [inline mpd/server.py]: Same, why the need for `str` ?

File: mpd/server.py [hunk 8]
```diff
@@ -120,7 +125,7 @@ def rescan_errored(mpd_root):
     errors = cur.fetchall()
     # Rerun blissify on them
     if errors is not None:
-        subprocess.check_call(["blissify", mpd_root] + errors)
+       subprocess.check_call(["blissify", mpd_root] + errors)
```
Comment (Phyks) [inline mpd/server.py]: Missing one whitespace.

File: mpd/server.py [hunk 9]
```diff
@@ -133,16 +138,19 @@ def update_db(mpd_root):
...
     except FileNotFoundError:
-        pass
+        logging.error("latest_mtime.txt file not found. Please call --full-rescan before --update.")
+        return
     songs = [x["file"] for x in client.find("modified-since", latest_mtime)]
     subprocess.check_call(["blissify", mpd_root] + songs)
     for song in songs:
-        last_modified = client.find("file", song)["last_modified"]
+        close_connection(client)
+        client = init_connection()
```
Comment (Phyks) [inline mpd/server.py]: Why closing and reopening the connection every time?

**Review by Phyks — COMMENTED** [review]
I could not test the PR for now. As long as it is working, I am ok with it to be merged. Just made a few comments about style.

File: mpd/server.py (hunk 7 above)
Comment (Polochon-street) [reply mpd/server.py]: write() takes a string as an argument, hence the str() (latest_mtime is an int); I believe here it is useful

File: mpd/server.py (hunk 6 above)
Comment (Polochon-street) [reply mpd/server.py]: Yes.

File: mpd/server.py (hunk 5 above)
Comment (Polochon-street) [reply mpd/server.py]: To mimick the fh.write() change, but here it is not needed - sorry

File: mpd/server.py (hunk 9 above)
Comment (Polochon-street) [reply mpd/server.py]: After some seconds without queries (which are long elapsed after the analysis process), the client timeouts; there must exist a cleaner way to enable the connection again, though.

File: mpd/server.py (hunk 5 above)
Comment (Phyks) [reply mpd/server.py]: Np.

File: mpd/server.py (hunk 6 above)
Comment (Phyks) [reply mpd/server.py]: ^^

File: mpd/server.py (hunk 7 above)
Comment (Phyks) [reply mpd/server.py]: Ah, indeed.

File: mpd/server.py (hunk 9 above)
Comment (Phyks) [reply mpd/server.py]: Ok, this should be refactored anyways #11 

We can stick with this and edit it later.
